server/lib/file-upload/aws.ts

```diff
@@ -61,16 +61,21 @@ export default class Aws implements FileStore {
     // DO Spaces use endpoints
     if (endpoint) {
       // AWS SDK v3 constructs a new URL() out of this, so it needs to have a "https://" prefix.
-      options.endpoint = endpoint
+      const endpointWithPrefix = endpoint.startsWith('https://') ? endpoint : `https://${endpoint}`
```

Comment:
I think you should make this more general in case we need to set a different endpoint for testing, etc. Just check if it matches `/^[^:/]+:\/\//`